ledger-core/virtual/integration/constructor_test.go

```diff
@@ -292,3 +294,128 @@ func TestVirtual_Constructor_NoVFindCallRequestWhenMissing(t *testing.T) {
 
 	mc.Finish()
 }
+
+// A.New calls B.New
+func TestVirtual_CallConstructorFromConstructor(t *testing.T) {
+	t.Log("C5090")
+
+	mc := minimock.NewController(t)
+
+	server, ctx := utils.NewUninitializedServer(nil, t)
+	defer server.Stop()
+
+	runnerMock := logicless.NewServiceMock(ctx, mc, func(execution execution.Context) string {
+		return execution.Request.Callee.String()
+	})
+	server.ReplaceRunner(runnerMock)
+	server.Init(ctx)
+
+	typedChecker := server.PublisherMock.SetTypedChecker(ctx, mc, server)
+
+	var (
+		isolation = contract.ConstructorIsolation()
+
+		classA        = gen.UniqueReference()
+		outgoingA     = server.RandomLocalWithPulse()
+		objectAGlobal = reference.NewSelf(outgoingA)
+
+		classB        = gen.UniqueReference()
+		objectBGlobal = reference.NewSelf(server.RandomLocalWithPulse())
+	)
+
+	// add ExecutionMocks to runnerMock
+	{
+		outgoingCall := executionevent.NewRPCBuilder(gen.UniqueReference(), classA).CallConstructor(classB, "New", []byte("123"))
+		objectAResult := requestresult.New([]byte("finish A.New"), objectAGlobal)
+		objectAResult.SetActivate(reference.Global{}, classA, []byte("state A"))
+		objectAExecutionMock := runnerMock.AddExecutionMock(classA.String())
+		objectAExecutionMock.AddStart(
+			func(ctx execution.Context) {
+				t.Log("ExecutionStart [A.New]")
+				require.Equal(t, classA, ctx.Request.Callee)
+				require.Equal(t, outgoingA, ctx.Request.CallOutgoing)
+			},
+			&executionupdate.ContractExecutionStateUpdate{
+				Type:     executionupdate.OutgoingCall,
+				Error:    nil,
+				Outgoing: outgoingCall,
+			},
+		)
+		objectAExecutionMock.AddContinue(
+			func(result []byte) {
+				t.Log("ExecutionContinue [A.New]")
+				require.Equal(t, []byte("finish B.New"), result)
+			},
+			&executionupdate.ContractExecutionStateUpdate{
+				Type:   executionupdate.Done,
+				Result: objectAResult,
+			},
+		)
+
+		objectBResult := requestresult.New([]byte("finish B.New"), objectBGlobal)
+		objectBResult.SetActivate(reference.Global{}, classB, []byte("state B"))
+		runnerMock.AddExecutionMock(classB.String()).AddStart(
+			func(ctx execution.Context) {
+				t.Log("ExecutionStart [B.New]")
+				require.Equal(t, classB, ctx.Request.Callee)
+				require.Equal(t, classA, ctx.Request.Caller)
+				require.Equal(t, []byte("123"), ctx.Request.Arguments)
+			},
+			&executionupdate.ContractExecutionStateUpdate{
+				Type:   executionupdate.Done,
+				Result: objectBResult,
+			},
+		)
+	}
+
+	// add checks to typedChecker
+	{
+		typedChecker.VCallRequest.Set(func(request *payload.VCallRequest) bool {
+			require.Equal(t, classB, request.Callee)
+			require.Equal(t, classA, request.Caller)
+			require.Equal(t, []byte("123"), request.Arguments)
+			return true // resend
+		})
+		typedChecker.VCallResult.Set(func(res *payload.VCallResult) bool {
+			switch res.Callee {
```

Comment:
this is not correct check. Callee in VCallResult should be object reference even if it's a constructor class!

Reply:
done